Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/GamePropertiesDialog.java

```diff
@@ -23,6 +22,8 @@
 
 public class GamePropertiesDialog extends DialogFragment
 {
+  private boolean mHadGameProfiles = false;
```

Comment:
This shouldn't need to be a member variable. You can store the return value of `recursivelyDeleteGameProfiles` as a local variable in `clearGameSettings`.

Reply:
I think there is a problem with converting the member variable to a local variable in `recursivelyDeleteGameProfiles`. Initializing the local variable immediately will clobber value changes during the loop. Even if I initialize the local variable later on, I don't know if there's a clean way to check if a file is a game profile, set `hadGameProfiles = true` if game file, then evaluate a potentially uninitialized `hadGameProfiles` by setting it to `false` if `hadGameProfiles` isn't `true`.

Reply:
You should be able to write `recursivelyDeleteGameProfiles` like this:

```
  private boolean recursivelyDeleteGameProfiles(File file, String gameId)
  {
    boolean hadGameProfiles = false;
    if (file.isDirectory())
    {
      try
      {
        for (File child : file.listFiles())
        {
          if (child.getName().startsWith(gameId) && child.isFile())
          {
            child.delete();
            hadGameProfiles = true;
          }
          hadGameProfiles |= recursivelyDeleteGameProfiles(child, gameId);
        }
      }
      catch (Exception ignored)
      {
      }
    }
    return hadGameProfiles;
  }
```